### PR TITLE
Allow Multiple Directories To Be Used In ALC

### PR DESCRIPTION
**Pull request by Matt-Cumber**
**Description of work.**
Removed checks for same directory in PABLV. The work for displaying a warning instead of an error is low priority and so has been excluded from this pr given the amount of work required to make that work.

**To test:**
Open Muon ALC
Then try to load runs from different directories e.g. MUSR 15189, MUSR62260
It should say multiple directories now instead of using the first one
Everything should function as before

Fixes part of #30747 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by robertapplin — CHANGES_REQUESTED** [review]
This works well and its nice to see some more tests, good job! Just a few comments

File: qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp [hunk 1]
```diff
@@ -350,6 +348,21 @@ void ALCDataLoadingPresenter::updateAvailableInfo() {
...
+  for (auto path : files) {
+    auto currentPath = path.substr(0, path.find_last_of("/\\"));
+    if (returnPath.empty())
+      returnPath = currentPath;
+    else if (currentPath != returnPath) {
+      returnPath = "Multiple Directories";
+      break; // End loop no need to check anymore
+    }
+  }
+  return returnPath;
```
[the remark this answers is not in the document]
Comment (Matt-Cumber) [reply qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp]: I've never seen use of all_of like this. Thanks for showing me

File: qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp [hunk 2]
```diff
@@ -348,19 +348,15 @@ void ALCDataLoadingPresenter::updateAvailableInfo() {
...
-      returnPath = currentPath;
-    else if (currentPath != returnPath) {
-      returnPath = "Multiple Directories";
-      break; // End loop no need to check anymore
-    }
-  }
-  return returnPath;
+std::string ALCDataLoadingPresenter::getPathFromFiles() const {
+  const auto files = m_view->getFiles();
+  const auto first_directory = files[0u].substr(0u, files[0u].find_last_of("/\\"));
```
Comment (robertapplin) [inline qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp:353]: Thanks for making those changes. Could we use camel case for this variable and then I'll approve this

Comment (Matt-Cumber) [reply qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp:353]: Oh whoops, been doing too much python recently

File: qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp [hunk 3]
```diff
@@ -350,6 +348,19 @@ void ALCDataLoadingPresenter::updateAvailableInfo() {
   m_numDetectors = ws->getInstrument()->getNumberDetectors();
 }
 
+std::string ALCDataLoadingPresenter::getPathFromFiles() const {
+  const auto files = m_view->getFiles();
+  if (files.empty())
```
Comment (robertapplin) [inline qt/scientific_interfaces/Muon/ALCDataLoadingPresenter.cpp:353]: Good catch 👍

**Review by robertapplin — APPROVED** [review]
Thanks for making those changes, this looks good to go

**Pull request merged**